src/pulselistener/pulselistener/lib/bus.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+import asyncio
+import inspect
+import multiprocessing
+
+from cli_common.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class MessageBus(object):
+    '''
+    Communication bus between processes
+    '''
+    def __init__(self, max_messages=None):
+        self.queues = {}
+        self.max_messages = max_messages
+        self.nb_messages = 0
+
+    def add_queue(self, name, mp=False):
+        '''
+        Create a new queue on the message bus
+        * asyncio by default
+        * multiprocessing when mp=True
+        '''
+        assert name not in self.queues, 'Queue {} already setup'.format(name)
+        if mp:
+            self.queues[name] = multiprocessing.Queue()
+        else:
+            self.queues[name] = asyncio.Queue()
+
+    def is_alive(self):
+        '''
+        Helper for unit tests runtimes
+        '''
+        if self.max_messages is None:
+            return True
+        return self.nb_messages <= self.max_messages
+
+    async def send(self, name, payload):
+        '''
+        Send a message on a specific queue
+        '''
+        assert name in self.queues, 'Missing queue {}'.format(name)
+        queue = self.queues[name]
+        if isinstance(queue, asyncio.Queue):
+            await queue.put(payload)
+        else:
+            queue.put(payload)
+            await asyncio.sleep(0)
+        self.nb_messages += 1
+
+    async def receive(self, name):
+        '''
+        Wait for a message on a specific queue
+        This is a blocking operation
+        '''
+        assert name in self.queues, 'Missing queue {}'.format(name)
+        queue = self.queues[name]
+        if isinstance(queue, asyncio.Queue):
+            return await queue.get()
+        else:
+            return queue.get()
```

Comment:
I wonder if we shouldn't use `await run_in_executor` so that while we wait for the message, other asyncio tasks can keep running

Reply:
https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.loop.run_in_executor